### PR TITLE
Fixed filter for volume's brick detail view

### PR DESCRIPTION
**Pull request by gnehapk**
Bugzilla - 1577938
tendrl-bug-id - https://github.com/Tendrl/ui/issues/964

Signed-off-by: Neha Gupta <gnehapk@gmail.com>

### CODE REVIEW

File: src/modules/bricks/volume-bricks/volume-bricks.js [hunk 1]
```diff
@@ -278,65 +279,93 @@
...
+            var subVol;
 
             filters.forEach(function(filter) {
-                if (!_matchesFilter(item, filter)) {
-                    matches = false;
-                    return false;
+                subVol = _matchesFilter(item, filter);
+
+                if (subVol === -9999) {
+                    return -9999;
```
Comment (a2batic) [inline src/modules/bricks/volume-bricks/volume-bricks.js]: @gnehapk, we can directly return subVol here, we need not check it

Comment (gnehapk) [reply src/modules/bricks/volume-bricks/volume-bricks.js]: @a2batic agreed.

**Comment by r0h4n** [comment]
@nthomas-redhat @gnehapk 

Every UI PR needs to be reviewed and approved by @cloudbehl  as well. 

@cloudbehl  please review

**Review by cloudbehl — APPROVED** [review]
Looks good to me!

**Review by a2batic — DISMISSED** [review]

File: src/modules/bricks/volume-bricks/volume-bricks.js [hunk 2]
```diff
@@ -278,65 +279,89 @@
...
+                subVol = _matchesFilter(item, filter);
             });
-            return matches;
+
+            return subVol;
         }
 
         function _applyFilters(filters) {
+            var subVol,
+                list = JSON.parse(JSON.stringify(vm.subVolumeList));;
```
Comment (cloudbehl) [inline src/modules/bricks/volume-bricks/volume-bricks.js]: I just saw there are two semi-colon on this line, can you correct it

**Review by cloudbehl — CHANGES_REQUESTED** [review]
can you fix the https://app.codacy.com/app/openstack/ui/pullRequest?prid=1671267 error whatever is possible?

**Review by cybernth — DISMISSED** [review]

**Review by a2batic — APPROVED** [review]

**Review by cloudbehl — APPROVED** [review]

**Review by cybernth — APPROVED** [review]

**Pull request merged**